toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -781,36 +782,107 @@ func calculateTotalPackages(packages []string, installRoot string) (installedPac
 	return
 }
 
-// addMachineID creates the /etc/machine-id file in the installChroot, if it is
-// not already present (systemd package will include it if installed).
-func addMachineID(installChroot *safechroot.Chroot) (err error) {
-	// From https://www.freedesktop.org/software/systemd/man/machine-id.html:
+// clearSystemdState clears the systemd state files that should be unique to each instance of the image. This is
+// based on https://systemd.io/BUILDING_IMAGES/. Primarily, this function will ensure that /etc/machine-id is configured
+// correctly, and that random seed and credential files are removed if they exist.
+// - installChroot is the chroot to modify
+// - enableSystemdFirstboot will set the machine-id file to "uninitialized" if true, and "" if false
+func clearSystemdState(installChroot *safechroot.Chroot, enableSystemdFirstboot bool) (err error) {
```

Comment:
is the capitalization of the first letter intentionally different from the systemconfig option? `enableSystemdFirstboot` vs `EnableSystemdFirstboot`

Reply:
Go enforces capitalized names for public values in packages and structs, but generally variables should be lower case. We treat our config structures as publicly accessible, hence `config.EnableSystemdFirstboot`, but the parameter for the function is not a "public" element so `enableSystemdFirstboot`.